examples/sagaMonitor/modules/logSaga.js

```diff
@@ -0,0 +1,156 @@
+/* eslint-disable no-console */
+import { is } from 'redux-saga/utils'
+import { effectTypes } from 'redux-saga/effects'
+import { consoleGroup, consoleGroupEnd } from './consoleGroup'
+import { CANCELLED, REJECTED } from './constants'
+import DescriptorFormatter from './DescriptorFormatter'
+
+export default function logSaga(manager) {
+  if (manager.getRootIds().length === 0) {
+    console.log('Saga monitor: No effects to log')
+  }
+  console.log('')
+  console.log('Saga monitor:', Date.now(), new Date().toISOString())
+  for (const id of manager.getRootIds()) {
+    logEffectTree(manager, id)
+  }
+  console.log('')
+}
+
+function logEffectTree(manager, effectId) {
+  const desc = manager.get(effectId)
+  const childIds = manager.getChildIds(effectId)
+
+  const formatter = getFormatterFromDescriptor(desc)
+  if (childIds.length === 0) {
+    console[formatter.logMethod](...formatter.getLog())
+  } else {
+    consoleGroup(...formatter.getLog())
+    for (const id of childIds) {
+      logEffectTree(manager, id)
+    }
+    consoleGroupEnd()
+  }
+}
+
+function getFormatterFromDescriptor(desc) {
+  const isCancel = desc.status === CANCELLED
+  const isError = desc.status === REJECTED
+
+  const formatter = new DescriptorFormatter(isCancel, isError)
+
+  const winnerInd = desc.winner ? (isError ? '✘' : '✓') : ''
+  formatter.addLabel(winnerInd).addLabel(desc.label)
+
+  if (is.iterator(desc.effect)) {
+    formatter.addValue(desc.effect.name).addDescResult(desc, true)
+  } else if (is.promise(desc.effect)) {
+    formatter
+      .addEffectType('promise')
+      .resetStyle()
```

Comment:
why we need to call `resetStyle` manually for all the cases except `addLabel`?

Reply:
This is adapted from the old code... Let me find a screenshot for you.